FirebaseTestProject/Views/PostRow.swift

```diff
@@ -8,23 +8,79 @@
 import SwiftUI
 
 struct PostRow: View {
-    let post: Post
+    @Binding  var post: Post
+    @State var isPresenting: Bool = false
+    @State private var showAlert: Bool = false
+    @State private var systemImage: String = ""
+    let deletePostAction:((Post) -> Void)
+    
+    init(post: Binding<Post>, deletePostAction: @escaping (Post) -> Void) {
+        self._post = post
+        self.deletePostAction = deletePostAction
+    }
+    
     var body: some View {
-        VStack {
-            Text(post.title)
-                .font(.title)
-                .padding(.bottom, 8)
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                Text(post.title)
+                    .font(.largeTitle)
+                    .padding(.leading)
+                Spacer()
+                let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
+                if userid == post.authorid {
+                    Button(action: {
+                        showAlert = true
+                    }, label: {
+                        Label("", systemImage: "trash")
+                    })
+                    .buttonStyle(PlainButtonStyle())
+                }
+                .padding()
+                .buttonStyle(PlainButtonStyle())
```

Comment:
> Cannot infer contextual base in reference to member 'padding'

These modifiers don’t compile because they’re attached to a conditional statement, not a SwiftUI element. You can delete these two lines. Then, if you’d like, you can add `.padding()` to the button above the other `buttonStyle` modifier you have here.

Reply:
Yeah, I erm...tried copy pasta since my XCode wasn't working and I was fully incapable of completing that task without an IDE telling me I'm not good at it :P no worries, it is fixed!